internal/sinks/postgres.go

```diff
@@ -545,13 +546,16 @@ func (pgw *PostgresWriter) MaintainUniqueSources() {
 		return
 	}
 
-	for _, tableName := range allDistinctMetricTables {
+	for i, tableName := range allDistinctMetricTables {
+		allDistinctMetricTables[i] = strings.Replace(tableName, "public.", "", 1)
+	}
+
+	for _, metricName := range allDistinctMetricTables {
 		foundDbnamesMap := make(map[string]bool)
 		foundDbnamesArr := make([]string, 0)
-		metricName := strings.Replace(tableName, "public.", "", 1)
 
 		logger.Debugf("Refreshing all_distinct_dbname_metrics listing for metric: %s", metricName)
-		rows, _ := pgw.sinkDb.Query(pgw.ctx, fmt.Sprintf(sqlDistinct, tableName, tableName))
+		rows, _ := pgw.sinkDb.Query(pgw.ctx, fmt.Sprintf(sqlDistinct, metricName, metricName))
```

Comment:
This is a bug! You cannot use trimmed table name in `FROM` clause hoping that `public` will be the first schema in search path. Details: https://www.postgresql.org/docs/18/ddl-schemas.html#DDL-SCHEMAS-PATH

If attacker will execute something like
```sql
SET search_path TO myschema,"$user";
```
the code will query the tables from `myschema` only.

I don't understand what exactly you were trying to solve here?

Reply:
aha, ok. So the thing is `admin.get_top_level_metric_tables()` knows exactly (sic!) how to return table names. In most case we will have the output as yours, because on most systems `public` is included in `search_path`